public/app/features/datasources/components/picker/DataSourceList.tsx

```diff
@@ -1,15 +1,20 @@
 import { css, cx } from '@emotion/css';
```

Comment:
the changes in this file are:
 - Avoid conditional hooks
 - Refactored `toggleFavoriteDatasource` to avoid duplicated code with `EditDataSourceActions.tsx`